modules/sync/syncs/urlsynchronization.cpp

```diff
@@ -253,4 +285,103 @@ std::string UrlSynchronization::generateUid() {
+    std::string line;
+
+    file >> line;
+    // Update ossync files that does not have a version number to new format
+    if (line == SynchronizationToken) {
+        if (_secondsUntilResync == 0) {
+            return false; // Force download new file
+        }
+
+        // We must close file early because createSyncFile writes to it
+        file.close();
+        createSyncFile();
+        // File is valid until some date
+        return true; 
+    }
+    // Otherwise first line is the version number.
+    std::string ossyncVersion = line;
+
+    //Format of 1.0 ossync:
+    //Version number: e.g., 1.0
+    //Date that specifies how long the files are valid for in ISO8601 format
+    //Valid to: yyyy-mm-ddThr:mn:sc.xxx
+
+    if (ossyncVersion == "1.0") {
+        std::getline(file >> std::ws, line);
+        std::string& fileIsValidToDate = line;
+        double fileValidAsJ2000 = Time::convertTime(fileIsValidToDate);
+
+        std::string todaysDate = Time::currentWallTime();
+        double todaysDateAsJ2000 = Time::convertTime(todaysDate);
+
+        // Issue warning if file is kept but user changed setting to download on startup.
+        if ((fileValidAsJ2000 > todaysDateAsJ2000) && _secondsUntilResync == 0) {
+            LWARNING(fmt::format(
+                "{}: File is valid to {} but asset specifies SecondsUntilResync = {} "
+                "Did you mean to re-download the file? If so, remove file from sync "
+                "folder to resync",
+                _identifier,
+                fileIsValidToDate,
+                _secondsUntilResync
+            ));
+        }
+
+        // Returns true if date in ossync file is 'larger' than todays date,
+        // i.e. no sync needed.
+        return fileValidAsJ2000 > todaysDateAsJ2000;
+    }
+    else {
+        LERROR(fmt::format(
+            "{}: Unknown ossync version number read. "
+            "Got {} while {} and below are valid",
+            _identifier,
+            ossyncVersion,
+            OssyncVersionNumber
+        ));
+        _state = State::Rejected;
+    }
+
+    return false;
+}
+
+void UrlSynchronization::createSyncFile(bool isFullySynchronized) const {
+    std::filesystem::path dir = directory();
+    std::filesystem::create_directories(dir);
+
+    dir.replace_extension("ossync");
+    std::ofstream syncFile(dir, std::ofstream::out);
+
+    std::string currentTimeAsISO8601 = Time::currentWallTime();
+    double currentTimeAsJ2000 = Time::convertTime(currentTimeAsISO8601);
+    
+    // With the format YYYY-MM... any year thats larger than 4 digits throws an error
+    // Limit the future date to year 9999
+    double futureTimeAsJ2000 = std::min(
+        currentTimeAsJ2000 + _secondsUntilResync,
+        MaxDateAsJ2000
+    );
+       
+    std::string fileIsValidTo = SpiceManager::ref().dateFromEphemerisTime(
+        futureTimeAsJ2000,
+        "YYYY-MM-DDTHR:MN:SC.###"
+    );
+   
+    syncFile << fmt::format(
```

Comment:
Lots of stuff going on here. easy to miss the `<<`

Maybe make a variable:
```
    std::string msg = fmt::format("{}\n{}\n", OssyncVersionNumber, fileIsValidTo);
    syncFile << msg;
```
?